File: migration_project/ui_qt/docks/log_dock.py
"""Runtime log dock for the Qt preview workbench."""

from __future__ import annotations

from datetime import datetime

from PySide6.QtWidgets import QPlainTextEdit, QVBoxLayout, QWidget


class LogDock(QWidget):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.output = QPlainTextEdit()
        self.output.setReadOnly(True)

        layout = QVBoxLayout(self)
        layout.setContentsMargins(8, 8, 8, 8)
        layout.addWidget(self.output)

    def append(self, message: str) -> None:
        stamp = datetime.now().strftime("%H:%M:%S")
        self.output.appendPlainText(f"[{stamp}] {message}")

File: migration_project/cpp_qt/src/docks/LogDock.cpp
#include "LogDock.h"

#include <QVBoxLayout>

LogDock::LogDock(QWidget* parent)
    : QWidget(parent)
{
    m_output = new QPlainTextEdit(this);
    m_output->setReadOnly(true);

    auto* layout = new QVBoxLayout(this);
    layout->setContentsMargins(8, 8, 8, 8);
    layout->addWidget(m_output);
}

void LogDock::append(const QString& message) {
    QString stamp = QDateTime::currentDateTime().toString("hh:mm:ss");
    m_output->appendPlainText(QString("[%1] %2").arg(stamp, message));
}
